sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -870,6 +870,197 @@ def test_planar_joint_advanced():
+    assert S._rot_type.upper() == 'BODY'
+    assert S._rot_order == 123
+    assert _simplify_matrix(N.dcm(A) - N_R_A) == zeros(3)
+    assert A.ang_vel_in(N).to_matrix(A) == N_w_A
+    assert C.masscenter.vel(N).to_matrix(A) == N_v_Co
+    # Test change of amounts
+    N, A, P, C, Pint, Cint = _generate_body(True)
+    S = SphericalJoint('S', P, C, coordinates=[q0, q1, q2], speeds=[u0, u1, u2],
+                       parent_point=N.x + N.y, child_point=-A.y + A.z,
+                       parent_interframe=Pint, child_interframe=Cint,
+                       rot_type='BODY', amounts=(q1, q0, q2), rot_order=123)
+    switch_order = lambda expr: expr.xreplace(
+        {q0: q1, q1: q0, q2: q2, u0: u1, u1: u0, u2: u2})
+    assert S._rot_type.upper() == 'BODY'
+    assert S._rot_order == 123
+    assert _simplify_matrix(N.dcm(A) - switch_order(N_R_A)) == zeros(3)
+    assert A.ang_vel_in(N).to_matrix(A) == switch_order(N_w_A)
+    assert C.masscenter.vel(N).to_matrix(A) == switch_order(N_v_Co)
+    # Test different rot_order
+    N, A, P, C, Pint, Cint = _generate_body(True)
+    S = SphericalJoint('S', P, C, coordinates=[q0, q1, q2], speeds=[u0, u1, u2],
+                       parent_point=N.x + N.y, child_point=-A.y + A.z,
+                       parent_interframe=Pint, child_interframe=Cint,
+                       rot_type='BodY', rot_order='yxz')
+    assert S._rot_type.upper() == 'BODY'
+    assert S._rot_order == 'yxz'
+    assert _simplify_matrix(N.dcm(A) - Matrix([
+        [-sin(q0) * cos(q1), sin(q0) * sin(q1) * cos(q2) - sin(q2) * cos(q0),
+         sin(q0) * sin(q1) * sin(q2) + cos(q0) * cos(q2)],
+        [-sin(q1), -cos(q1) * cos(q2), -sin(q2) * cos(q1)],
+        [cos(q0) * cos(q1), -sin(q0) * sin(q2) - sin(q1) * cos(q0) * cos(q2),
+         sin(q0) * cos(q2) - sin(q1) * sin(q2) * cos(q0)]])) == zeros(3)
+    assert A.ang_vel_in(N).to_matrix(A) == Matrix([
+        [u0 * sin(q1) - u2], [u0 * cos(q1) * cos(q2) - u1 * sin(q2)],
+        [u0 * sin(q2) * cos(q1) + u1 * cos(q2)]])
+    assert C.masscenter.vel(N).to_matrix(A) == Matrix([
+        [-sqrt(2) * (u0 * sin(q2 + pi / 4) * cos(q1) + u1 * cos(q2 + pi / 4))],
+        [u0 * sin(q1) - u2], [u0 * sin(q1) - u2]])
+
+
+def test_spherical_joint_orient_space():
+    q0, q1, q2, u0, u1, u2 = dynamicsymbols('q0:3, u0:3')
+    N_R_A = Matrix([
+        [-sin(q0) * sin(q2) - sin(q1) * cos(q0) * cos(q2),
+         sin(q0) * sin(q1) * cos(q2) - sin(q2) * cos(q0), cos(q1) * cos(q2)],
+        [-sin(q0) * cos(q2) + sin(q1) * sin(q2) * cos(q0),
+         -sin(q0) * sin(q1) * sin(q2) - cos(q0) * cos(q2), -sin(q2) * cos(q1)],
+        [cos(q0) * cos(q1), -sin(q0) * cos(q1), sin(q1)]])
+    N_w_A = Matrix([
+        [u1 * sin(q0) - u2 * cos(q0) * cos(q1)],
+        [u1 * cos(q0) + u2 * sin(q0) * cos(q1)], [u0 - u2 * sin(q1)]])
+    N_v_Co = Matrix([
+        [u0 - u2 * sin(q1)], [u0 - u2 * sin(q1)],
+        [sqrt(2) * (-u1 * sin(q0 + pi / 4) + u2 * cos(q0 + pi / 4) * cos(q1))]])
```

Comment:
How confident are you that all these are correct?

Reply:
If I remember correctly I created this output using `orient_axis` (or `orient_body_fixed`), so in that sense quite confident, since I belief both of those methods to give correct output.

Reply:
But that's checking the result with the same solution method, no? I guess I mean more do you know these are correct by doing it by hand? Or some other checks. We got some of these wrong with Sudeep's work because we didn't have careful sanity checks.

Reply:
Overall I do most of the time assume that for checking that I can use at least one level lower API to generate the answer. This means that in cane of the PR around `orient_body_fixed` and `orient_space_fixed` that I verified those methods using matrix multiplication in sympy and `orient_axis`. As for the joints I've always use `orient_axis` and the velocity theorems to do all computations.

But if you like I can try and think of some more sanity checks.

Reply:
That's sufficient, just as long as you are being careful.